hpssa/hpssa.py

```diff
@@ -323,6 +333,10 @@ def _raw_system_info(self):
         adapters = parse_adapter_details(raw_details)
 
         for adapter in adapters:
+            if not all([k in adapter for k in
+                        ('slot', 'configuration', 'drives')]):
+                raise HPRaidException('Parsing adapters failed due to missing '
```

Comment:
Instead of raising here, lets set that adapters error state, log.error(), and move on.